lib/versioning/VersioningRequestProcessor.js

```diff
@@ -50,8 +50,8 @@ class VersioningRequestProcessor {
         this.repairing = {};
     }
 
-    generateVersionId() {
-        return genVID(this.uidCounter++);
+    generateVersionId(replicationGroupToken) {
+        return genVID(this.uidCounter++, replicationGroupToken);
```

Comment:
You have to pass the config value from S3 and metadata, take it as the input when instantiating this class, then set it as the global value to use thereafter.
This may be used by both metadata and bucketfile so no changes to the APIs down here is better.